RecoParticleFlow/PFClusterProducer/src/PFClusterEMEnergyCorrector.cc

```diff
@@ -11,261 +8,421 @@ namespace {
   bool sortByKey(const EEPSPair& a, const EEPSPair& b) {
     return a.first < b.first;
   } 
+
+  double getOffset(const double lo, const double hi) { return lo + 0.5*(hi-lo); }
+  double getScale(const double lo, const double hi) { return 0.5*(hi-lo); }
 }
 
 PFClusterEMEnergyCorrector::PFClusterEMEnergyCorrector(const edm::ParameterSet& conf, edm::ConsumesCollector &&cc) :
-  _calibrator(new PFEnergyCalibration) {
+  calibrator_(new PFEnergyCalibration) {
 
-   _applyCrackCorrections = conf.getParameter<bool>("applyCrackCorrections");
-   _applyMVACorrections = conf.getParameter<bool>("applyMVACorrections");
-   _maxPtForMVAEvaluation = conf.getParameter<double>("maxPtForMVAEvaluation");
-  
+   applyCrackCorrections_ = conf.getParameter<bool>("applyCrackCorrections");
+   applyMVACorrections_ = conf.getParameter<bool>("applyMVACorrections");
+   srfAwareCorrection_ = conf.getParameter<bool>("srfAwareCorrection");
+
+   maxPtForMVAEvaluation_ = conf.getParameter<double>("maxPtForMVAEvaluation");
+     
+   if (applyMVACorrections_) {
+
+     meanlimlowEB_ = -0.336;
+     meanlimhighEB_ = 0.916;
+     meanoffsetEB_ = getOffset(meanlimlowEB_, meanlimhighEB_);
+     meanscaleEB_ = getScale(meanlimlowEB_, meanlimhighEB_);
+
+     meanlimlowEE_ = -0.336;
+     meanlimhighEE_ = 0.916;
+     meanoffsetEE_ = getOffset(meanlimlowEE_, meanlimhighEE_);
+     meanscaleEE_ = getScale(meanlimlowEE_, meanlimhighEE_);
+     
+     sigmalimlowEB_ = 0.001;
+     sigmalimhighEB_ = 0.4;
+     sigmaoffsetEB_ = getOffset(sigmalimlowEB_, sigmalimhighEB_);
+     sigmascaleEB_ = getScale(sigmalimlowEB_, sigmalimhighEB_);
+     
+     sigmalimlowEE_ = 0.001;
+     sigmalimhighEE_ = 0.4;
+     sigmaoffsetEE_ = getOffset(sigmalimlowEE_, sigmalimhighEE_);
+     sigmascaleEE_ = getScale(sigmalimlowEE_, sigmalimhighEE_);
    
-  if (_applyMVACorrections) {
-    _recHitsEB = cc.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitsEBLabel"));
-    _recHitsEE = cc.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitsEELabel"));
+     recHitsEB_ = cc.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitsEBLabel"));
+     recHitsEE_ = cc.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitsEELabel"));
+     autoDetectBunchSpacing_ = conf.getParameter<bool>("autoDetectBunchSpacing");
+     
+     if (autoDetectBunchSpacing_) {
+       bunchSpacing_ = cc.consumes<unsigned int>(edm::InputTag("bunchSpacingProducer"));
+       bunchSpacingManual_ = 0;
+     } else {
+       bunchSpacingManual_ = conf.getParameter<int>("bunchSpacing");
+     }
+     
+     condnames_mean_25ns_.push_back("ecalPFClusterCorV2_EB_pfSize1_mean_25ns");
```

Comment:
hi @rafaellopesdesa - please change these push_backs to something more readable. i suggest

condnames_mean_25ns_.insert(condnames_mean_25ns_.end(), { "a", "b", "c"})